app/wama.web/Controllers/UserToolController.cs

```diff
@@ -173,10 +173,79 @@ public IActionResult UserAccountAddNewUser()
                     }
                 }
             }
-
             return View($"{Constants.ADMIN_CONSOLE_USER_TOOL_DIRECTORY}/UserAccountAddNewUser.cshtml");
         }
 
+        public IActionResult AddCertification(string memberId)
+        {
+            ViewBag.MemberId = memberId;
+            SetActiveConsoleTool(Constants.ADMIN_CONSOLE_USERS_PATRONS);
+            return View($"{Constants.ADMIN_CONSOLE_USER_TOOL_DIRECTORY}/AddCertification.cshtml");
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> AddCertification(CertificationViewModel Certification)
+        {
+            if (!ModelState.IsValid)
+            {
+                this.SetErrorMessages(ModelState.Values
+                    .Where(val => val.ValidationState == ModelValidationState.Invalid)
+                    .Select(val => val.Errors.FirstOrDefault().ErrorMessage));
+            }
+            else
+            {
+                //if certification type already exist then if not expired error if expired update
+                try
+                {
+                    await _CertificationService.AddCertificationAsync(Certification);
+                    return RedirectToAction(
+                        actionName: nameof(ViewCertification),
+                        routeValues: new
+                        {
+                            MemberId = Certification.MemberId
+                        });
+                }
+                catch (Microsoft.EntityFrameworkCore.DbUpdateException ex)
+                when (ex.InnerException is System.Data.SqlClient.SqlException)
+                {
+                    if (ex.InnerException.Message.Contains("Cannot insert duplicate key"))
+                    {
+                    }
+                    else
+                    {
+                    }
+                }
+                catch (Exception ex)
+                {
+                    var errMessages = new List<string>();
+                    while (ex != null)
+                    {
+                        errMessages.Add(ex.Message);
+                        ex = ex.InnerException;
+                    }
+
+                    SetErrorMessages(errMessages);
+                }
+            }
+
+            ViewBag.MemberId = Certification.MemberId;
+            return View(Certification);
```

Comment:
why are you returning View(Certification);

did you test this?
shouldn't you be specifying a view to return?